osutil/disks/disks.go

```diff
@@ -112,6 +112,13 @@ type Partition struct {
 	KernelDeviceNode string
 }
 
+// RootMountPointsForPartition returns all mounts from the mount table which are
+// for the root directory of the specified partition. The order in which they
+// are returned is the exact order that they appear in the mount table.
+func RootMountPointsForPartition(p Partition) ([]string, error) {
```

Comment:
I hope I'm not bikeshedding too much :-), but when I read this function name I read "this function returns the root mount points for the given partition", which is not waht the function returns. I'd suggest renaming it to

    MountPointsForPartitionRoot

I wish we could call it `MountPointsForPartition'sRoot` but something tells me that's not possible.

Reply:
ack, this makes some sense after discussing on IRC, I will rename in a followup I think

Reply:
done in https://github.com/snapcore/snapd/pull/10872